ResearchSuiteUI/ResearchSuiteUI/iOS/Step View Controllers/RSDOverviewStepViewController.swift

```diff
@@ -0,0 +1,260 @@
+//
+// 2.  Redistributions in binary form must reproduce the above copyright notice,
+// this list of conditions and the following disclaimer in the documentation and/or
+// other materials provided with the distribution.
+//
+// 3.  Neither the name of the copyright holder(s) nor the names of any contributors
+// may be used to endorse or promote products derived from this software without
+// specific prior written permission. No license is granted to the trademarks of
+// the copyright holders even if such marks are included in this software.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+import UIKit
+import UserNotifications
+
+/// `RSDOverviewStepViewController` is a customizable view controller that is designed to be the first view
+/// displayed for an active task that may require checking the user's permissions and allows the user to set
+/// a notification reminder to perform the task at a later time.
+open class RSDOverviewStepViewController: RSDStepViewController {
+    
+    /// Override viewDidAppear to setup notification handling.
+    open override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        
+        // Check the authorization status
+        _updateAuthorizationStatus()
+        
+        // If this is a reminder action then set that and keep a pointer to it.
+        self.reminderAction = self.action(for: .navigation(.skip)) as? RSDReminderUIActionObject
+        
+        // Remove any previous reminder.
+        if let reminderIdentifier = reminderAction?.reminderIdentifier {
+            UNUserNotificationCenter.current().removePendingNotificationRequests(withIdentifiers: [reminderIdentifier])
+        }
+    }
+    
+    
+    // MARK: Permission handling
+    
+    /// Override the forward button to disable until the status is checked.
+    override open var isForwardEnabled: Bool {
+        return super.isForwardEnabled && !(_authStatus?.isDenied() ?? true)
+    }
+    
+    private var _authStatus: RSDAuthorizationStatus?
+    
+    /// Check authorization status
+    private func _updateAuthorizationStatus() {
+        
+        // Check the permission status for all required permissions. This will not **request** permission,
```

Comment:
seems like it will, though, in one particular situation...?

Reply:
No, it will **only** check if the permission is now authorized if it was **previously** denied. In other words, if there isn't a cached value saying "last time we checked, you had denied permission" then it will not verify that value. So it only ever shows an alert if you have previously denied permission.